### PR TITLE
Ability to discard single lines

### PR DESCRIPTION
**Pull request by estib-vega**
## ☕️ Reasoning

This pull request adds the ability to discard single lines in the file hunk context menu. This feature allows users to selectively remove individual lines from the changes, providing more granular control over the commit process.

## 🧢 Changes

- Added a method to unapply ownership lines, enabling the determination of a subset of hunk lines to be discarded.
- Implemented the option to discard single lines in the file hunk context menu.

### CODE REVIEW

**Review by ndom91 — APPROVED** [review]
Works on my machine, nice!

File: crates/gitbutler-diff/src/diff.rs [hunk 1]
```diff
@@ -385,6 +385,227 @@ fn reverse_patch(patch: &BStr) -> Option<BString> {
...
+    Some(PatchHeaderInfo {
+        old_start,
+        old_lines,
+        new_start,
+        new_lines,
+        rest,
+    })
+}
+
+fn create_patch_line_map_key(old_line: Option<u32>, new_line: Option<u32>) -> String {
```
Comment (Byron) [inline crates/gitbutler-diff/src/diff.rs:469]: `(Option<u32>, Option<u32>)` will be perfectly suitable as hashmap key, without the need for allocation. Maybe there is other reasons to go with `String` though, but I think there shouldn't be.

**Review by Byron — COMMENTED** [review]
I just took a quick look and hope the comment will be useful.

Generally, when looking at this, I think there should be no need to deal with stringly-types like patches that could be applied with `patch`, and hope to one day be able to help remove it.

File: crates/gitbutler-diff/src/diff.rs (hunk 1 above)
Comment (estib-vega) [reply crates/gitbutler-diff/src/diff.rs:469]: Thanks for the comment!
Indeed there is no need for strings to be created. I'll add a follow-up PR with the improvement :)

**Pull request merged**